sql/sys_vars.cc

```diff
@@ -5565,6 +5585,35 @@ Sys_var_rpl_filter::global_value_ptr(THD *thd,
   return ret;
 }
 
+const uchar *
+Sys_var_binlog_filter::global_value_ptr(THD *thd,
+                                        const LEX_CSTRING *base_name) const
+{
+  char buf[256];
+  String tmp(buf, sizeof(buf), &my_charset_bin);
+  uchar *ret;
+
+  mysql_mutex_unlock(&LOCK_global_system_variables);
```

Comment:
You don't need to switch locks at all, holding the already attained `LOCK_global_system_variables` is safe.